### ISSUE TITLE
function truncate_graph_dist returns an unexpected result

### ISSUE THREAD

**Issue by guibar** [gboeing/osmnx#648]
I found a problem with this function whereby some unexpected nodes make it into the returned graph.
My debugger pointed me to the issue straight away. Starting on line 40 of truncate.py, we have:

```python
distances = nx.shortest_path_length(G, source=source_node, weight=weight)
distant_nodes = {k: v for k, v in distances.items() if v > max_dist}
G = G.copy()
G.remove_nodes_from(distant_nodes)
```

The problem is that nodes which are unreachable from the source (in my case, nodes which are on an incoming edge on the border of my zone of interest) make it into the copy of G because they are not in distant_nodes as a result of the shortest_path_length call not including them in the distances dict.

**Environment information**
  - I am on Linux, python = 3.8.5
  - osmnx= 1.0.1

Here is a reproducible example:

```python
import osmnx

d2_box = (1.932, 2.204, 49.323, 49.46)
G = osmnx.graph.graph_from_bbox(d2_box[3], d2_box[2], d2_box[1], d2_box[0],
                                network_type='drive', simplify=True, retain_all=False,
                                truncate_by_edge=True, clean_periphery=True)
GG = osmnx.truncate.truncate_graph_dist(G,7846489021,max_dist=20)
list(GG.nodes())

```
returns 
[5538515552, 265534270] which are nowhere near the source node.

**Comment by gboeing** [comment]
Thanks! Do you have a proposed fix you'd like to contribute as a PR?

**Comment by guibar** [comment]
Hi Geoff,

Thanks for all your work btw!
Yes, I can propose something although python is not my language of
expertise.
I am doing it right now.

Guillaume

**Comment by gboeing** [comment]
closed by #651
